pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -569,20 +570,35 @@ func (c *Controller) getProxyServiceInstancesByEndpoint(endpoints v1.Endpoints,
 					continue
 				}
 
-				// There is only one IP for kube registry
 				proxyIP := proxy.IPAddresses[0]
 
 				if hasProxyIP(ss.Addresses, proxyIP) {
 					out = append(out, c.getEndpoints(proxyIP, port.Port, svcPort, svc))
 				}
 
 				if hasProxyIP(ss.NotReadyAddresses, proxyIP) {
-					nrEP := c.getEndpoints(proxyIP, port.Port, svcPort, svc)
-					out = append(out, nrEP)
+					out = append(out, c.getEndpoints(proxyIP, port.Port, svcPort, svc))
 					if c.Env != nil {
 						c.Env.PushContext.Add(model.ProxyStatusEndpointNotReady, proxy.ID, proxy, "")
 					}
 				}
+
+				// appending the instances by changing the NetworkEndpoint.Address when there are multiple ips
+				for index, ip := range proxy.IPAddresses {
+					if index != 0 && len(out) > 0 {
+
+						if hasProxyIP(ss.Addresses, ip) {
+							out = append(out, c.overrideServiceIPAndNetwork(out[0], ip))
+						}
+
+						if hasProxyIP(ss.NotReadyAddresses, ip) {
+							out = append(out, c.overrideServiceIPAndNetwork(out[0], ip))
```

Comment:
I think it is also not right here in `getProxyServiceInstancesByEndpoint`

Reply:
Already modified